c2rust-analyze/src/main.rs

```diff
@@ -465,7 +517,23 @@ fn run(tcx: TyCtxt) {
                         }
                         _ => continue,
                     },
-                    // TODO: Rvalue::Cast (certain cases)
+                    Rvalue::Cast(_, ref op, _) => {
+                        if let Some(p) = op.place().filter(|p| acx.c_void_ptrs.contains(p)) {
+                            // This is a special case for types being casted from *libc::c_void to a pointer
+                            // to some other type, e.g. `let foo = malloc(..) as *mut Foo;`
+                            // carry over the pointer id of *libc::c_void, but match the pointer ids
+                            // of the casted-to-type for the rest
+                            acx.special_casts.insert(p, lhs);
+                        }
+
+                        if acx.c_void_ptrs.contains(&lhs) {
+                            // This is a special case for types being casted to *libc::c_void
+                            // acx.special_casts.insert(p, lhs);
```

Comment:
What does this mean?

Reply:
that line was left in by mistake. removed